Automattic/jetpack
# Publicize: fix refreshing social media connections

### Branched off from https://github.com/Automattic/jetpack/pull/21090

This PR fixes the issue when the user changes the social media connections from the dotcom admin page. The issue is described [here](https://github.com/Automattic/jetpack/issues/21093), but in short, it doesn't refresh the connections list once the user edits them.

Fixes https://github.com/Automattic/jetpack/issues/21093 

### About the approach

It consumes the [Page Visibility API](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API), through the [react-page-visibility](https://github.com/pgilad/react-page-visibility) React lib. It's pretty light and simplifies the implementation.

#### Changes proposed in this Pull Request:

* Publicize: fix refreshing social media connections

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Create a new post / Edit a post.
* Open the Jetpack sidebar.
* Click on the `Connect an account` external link.
* Edit the social media connections.
* Confirm the connections list is automatically updated once you go back to the block editor (it performs a request to the API, hitting the `wpcom/v2/publicize/connection-test-results` endpoint).
* Close the sidebar
* Change again the connections
* Confirm the browser doesn't hit the API
* Open the sidebar again
* Confirm the client hits the API
* Confirm the connections list is updated

The video below shows all these testing steps:

https://user-images.githubusercontent.com/77539/133631969-3d77b329-83fb-4f90-a41c-46d2df9ab149.mov
